server/src/main/java/org/jboss/as/server/deployment/DeploymentResourceSupport.java

```diff
@@ -82,6 +82,47 @@ public ModelNode registerDeploymentSubsystemResource(final String subsystemName,
     }
 
     /**
+     * Checks to see if a resource has already been registered for the specified address on the subsystem.
+     *
+     * @param subsystemName the name of the subsystem
+     * @param address       the address to check
+     *
+     * @return {@code true} if the address exists on the subsystem otherwise {@code false}
+     */
+    public boolean hasDeploymentSubModel(final String subsystemName, final PathElement address) {
+        final Resource root = deploymentUnit.getAttachment(DEPLOYMENT_RESOURCE);
+        final PathElement subsystem = PathElement.pathElement(SUBSYSTEM, subsystemName);
+        return root.hasChild(subsystem) && root.getChild(subsystem).hasChild(address);
```

Comment:
Probably actually. Since you can register subsystem resources there should be a check for that as well.